extensions/reviewed/FPS.json

```diff
@@ -19,82 +28,263 @@
     {
-      "description": "Return your game's current FPS count.",
-      "fullName": "FPS (Frames per second)",
+      "fullName": "",
+      "functionType": "Action",
+      "name": "onSceneLoaded",
+      "sentence": "",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::Comment",
+          "color": {
+            "b": 109,
+            "g": 230,
+            "r": 255,
+            "textB": 0,
+            "textG": 0,
+            "textR": 0
+          },
+          "comment": "Save starting time and set default value"
+        },
+        {
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [],
+          "actions": [
+            {
+              "type": {
+                "value": "FPS::RestartFpsCounter"
+              },
+              "parameters": [
+                "",
+                "",
+                ""
+              ]
+            },
+            {
+              "type": {
+                "value": "ModVarScene"
+              },
+              "parameters": [
+                "__FPS.UpdatePeriod",
+                "=",
+                "1"
+              ]
+            }
+          ]
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "fullName": "",
+      "functionType": "Action",
+      "name": "onScenePostEvents",
+      "sentence": "",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [],
+          "actions": [
+            {
+              "type": {
+                "value": "ModVarScene"
+              },
+              "parameters": [
+                "__FPS.FramesCounted",
+                "+",
+                "1"
+              ]
+            }
+          ]
+        },
+        {
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [
+            {
+              "type": {
+                "value": "BuiltinCommonInstructions::CompareNumbers"
+              },
+              "parameters": [
+                "Time(\"timestamp\") - Variable(__FPS.StartTime)",
+                ">",
+                "1000 * Variable(__FPS.UpdatePeriod) "
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "type": {
+                "value": "ModVarScene"
+              },
+              "parameters": [
+                "__FPS.FramesPerSecond",
+                "=",
+                "1000 * Variable(__FPS.FramesCounted) / (Time(\"timestamp\") - Variable(__FPS.StartTime))"
+              ]
+            },
+            {
+              "type": {
+                "value": "FPS::RestartFpsCounter"
+              },
+              "parameters": [
+                "",
+                ""
+              ]
+            }
+          ]
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Average frames per second (FPS) during the last update period.",
```

Comment:
```suggestion
      "description": "Average frames per second (FPS) during the last update period. When measuring a game performance, rendering time should be used instead of FPS because going from 100 to 80 FPS is 2,5 ms difference where going from 40 to 20 FPS is 25 ms difference.
```

Reply:
I guess we could do a similar "AverageRenderTime" that works similar to FPS (taking an average of the measuring period).  Do you think that would be useful?

Reply:
No, you can't measure the render time with events because it measure the waiting time too.
GDevelop or browser profilers must be used.